gdal/ogr/ogrsf_frmts/dxf/ogrdxfreader.cpp

```diff
@@ -179,13 +177,57 @@ int OGRDXFReader::ReadValueRaw( char *pszValueBuf, int nValueBufSize )
            && achSrcBuffer[iEOL] != '\0' )
         iEOL++;
 
-    if( achSrcBuffer[iEOL] == '\0' )
-        return -1;
+    while( achSrcBuffer[iEOL] == '\0' )
+    {
+        // The line is longer than the buffer. Let's copy what we have so
+        // far into our string, and read more
+        const auto nValueLength = osValue.length();
+        osValue.resize( nValueLength + iEOL - iSrcBufferOffset, '\0' );
```

Comment:
couldn't this grow out of control (thinking of potential hostile files) ? Shoudn't we had a check up to ~ 1MB of osValue.size() ?

Reply:
The code now bails out if osValue is going to exceed 1MB.